cli/cli.go

```diff
@@ -41,15 +45,30 @@ func (f *Flags) Init(all ...Initializer) error {
 		}
 	}
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 	if f.cpuprofile != "" {
 		f.runCPUProfile(f.cpuprofile)
 	}
-	return nil
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	var cleanup = func() {
+		cancel()
+		f.cleanup()
+	}
+	return &interruptedContext{ctx}, cleanup, nil
+}
+
+type interruptedContext struct{ context.Context }
+
+func (s *interruptedContext) Err() error {
+	err := s.Context.Err()
+	if errors.Is(err, context.Canceled) {
+		return errors.New("interrupted")
+	}
+	return err
 }
```

Comment:
I'm surprised to see this here. Does it need to move? If not, I think it should stay where it is.

Reply:
Here's the story: I moved lake flags into cmd/zed/root command, zq uses this command and so now zq had the lake flags. zq only uses the root command because it has the convenient context/signal init stuff. It seem sensible to put this in cli.Flags since all our commands use this and also need ctx. I could move this to another struct, maybe something like cli.Base, and maybe that is the right thing to do, but I wanted to avoid adding another thing.